docs/tutorials/register-a-plugin/arc.md

```diff
@@ -0,0 +1,120 @@
+- _voteParamsHash: Voting machine parameters.
+
+Beside of these parameters, you can pass as many as you want; in case you want to store any useful information in a contract's variable.
+
+Next, you will need to create a struct `Proposal`; this will allow you to store the information needed for the proposal (i.e: tokens to transfer/receive, REP to give, and any other use case you can imagine), note that this is the information the contract needs to have access to, things like title or description will be stored on IPFS (when we add our plugin at Subgraph level). Also, we will need a `proposals` mapping, to identify every proposal with a unique ID. Example with `TokenTrade` plugin:
+
+```
+struct Proposal {
+    address beneficiary;
+    IERC20 sendToken;
+    uint256 sendTokenAmount;
+    IERC20 receiveToken;
+    uint256 receiveTokenAmount;
+    bool passed;
+    bool decided;
+}
+
+mapping(bytes32=>Proposal) public proposals;
+```
+
+Create a `proposal${PLUGIN_NAME}`, (i.e: `proposalTokenTrade`). This function will create the proposal in the Voting Machine, by calling the `votingMachine.propose(2, ...)` function, the first argument of this function always needs to be 2 because it's the number of choices a proposal will have (Yes or No). Check the implementation of `TokenTrade`: 
+```
+function proposeTokenTrade(
+    IERC20 _sendToken,
+    uint256 _sendTokenAmount,
+    IERC20 _receiveToken,
+    uint256 _receiveTokenAmount,
+    string memory _descriptionHash
+)
+public
+returns(bytes32 proposalId)
+{
+    require(
+        address(_sendToken) != address(0) && address(_receiveToken) != address(0),
+        "Token address must not be null"
+    );
+    require(_sendTokenAmount > 0 && _receiveTokenAmount > 0, "Token amount must be greater than 0");
+
+    _sendToken.safeTransferFrom(msg.sender, address(this), _sendTokenAmount);
+    proposalId = votingMachine.propose(2, voteParamsHash, msg.sender, address(avatar));
+
+    proposals[proposalId] = Proposal({
+        beneficiary: msg.sender,
+        sendToken: _sendToken,
+        sendTokenAmount: _sendTokenAmount,
+        receiveToken: _receiveToken,
+        receiveTokenAmount: _receiveTokenAmount,
+        passed: false,
+        decided: false
+    });
+
+    proposalsBlockNumber[proposalId] = block.number;
+
+    emit TokenTradeProposed(
+        address(avatar),
+        proposalId,
+        _descriptionHash,
+        msg.sender,
+        _sendToken,
+        _sendTokenAmount,
+        _receiveToken,
+        _receiveTokenAmount
+    );
+}
+```
+
+In order to execute the proposal once it has passed, you need to implement an `executeProposal` method like this:
+```
+function executeProposal(bytes32 _proposalId, int256 _decision)
+external
+onlyVotingMachine(_proposalId)
+override
+returns(bool) {
+    ...
+}
+```
+In this you will define what happens once the proposal has been approved or rejected, based on the `_decision` parameter.
```

Comment:
What about adding the TokenTrade example here too? Maybe should also note that execution logic should usually be separated into a different function, whereas this one should be used just to mark a proposal passed.